lib/streamlit/runtime/fragment.py

```diff
@@ -119,43 +119,115 @@ def fragment(
     *,
     run_every: int | float | timedelta | str | None = None,
 ) -> Callable[[F], F] | F:
-    """Allow a function to be run independently of the full script.
-
-    Functions decorated with ``@st.experimental_fragment`` are handled specially within
-    an app: when a widget created within an invocation of the function (a fragment) is
-    interacted with, then only that fragment is rerun rather than the full streamlit app.
+    """Decorator to turn a function into a fragment which can rerun independently\
+    of the full script.
+
+    When a user interacts with an input widget created by a fragment, Streamlit
+    only reruns the fragment instead of the full script. If ``run_every`` is set,
+    Streamlit will also rerun the fragment at the specified interval while the
+    session is active, even if the user is not interacting with your app.
+
+    To trigger a full script rerun from inside a fragment, call ``st.rerun()``
+    directly. Any values from the fragment that need to be accessed from
+    the wider app should generally be stored in Session State.
+
+    When Streamlit element commands are called directly in a fragment, the
+    elements are cleared and redrawn on each fragment rerun, just like all
+    elements are redrawn on each full-script rerun. The rest of the app is
+    persisted during a fragment rerun. When a fragment renders elements into
+    externally created containers, the elements will not be cleared with each
+    fragment rerun. In this case, elements will be drawn accumulatively in
+    those containers with each fragment rerun, until the next full-script
+    rerun.
+
+    Calling `st.sidebar` in a fragment is not supported. To write elements to
+    the sidebar with a fragment, use `with st.sidebar` outside of the fragment
+    function call.
```

Comment:
nit: would "wrap the fragment function call with `with st.sidebar:`" be slightly clearer / more specific? But I think this version is OK too.